include/IECoreScene/private/PrimitiveAlgoUtils.h

```diff
@@ -102,17 +101,52 @@ template< typename T > struct IsArithmeticVectorTypedData
 
 struct AverageValueFromVector
 {
-	typedef IECore::DataPtr ReturnType;
+	AverageValueFromVector()
+	{
+	}
 
-	template<typename From> ReturnType operator()( typename From::ConstPtr data )
+	template<typename T>
+	IECore::DataPtr operator()( const IECore::TypedData<std::vector<T> > *data )
```

Comment:
You should be able to use `enable_if` with `IsArithmeticVectorTypedData` to limit matches to the ones your want, and then you won't need all the overloads below, just a final one taking `Data *`. Example here :
https://github.com/ImageEngine/cortex/blob/master/src/IECore/ObjectInterpolator.cpp#L91-L92

Reply:
Thanks, that's exactly what I was missing.

Reply:
Addressed in ca0117e